pr-dhl-woocommerce/includes/checkout-blocks/dhl-closest-drop-point/block.js

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useState, useCallback } from '@wordpress/element';
+import { Notice } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { CART_STORE_KEY } from '@woocommerce/block-data';
+import { debounce } from 'lodash';
+import { __ } from '@wordpress/i18n';
+
+export const Block = ({ checkoutExtensionData }) => {
+    const { setExtensionData } = checkoutExtensionData;
+
+    // Access the localized data from prDhlGlobals
+    const imgUrl      = prDhlGlobals?.pluginUrl + "/assets/img/dhl-official.png";
+    const dhlSettings = prDhlGlobals.dhlSettings;
+
+    const [error, setError] = useState( null );
+    const [closestDP, setClosestDP] = useState('no');
+    const [displayClosest, setDisplayClosest] = useState( true );
+
+    const closestAvailable = dhlSettings?.closest_drop_point;
+    const validCountries   = [ 'SE', 'FI', 'BE', 'AT', 'FR' ];
```

Comment:
this should be extracted to the backend end instead 
https://github.com/shadimanna/dhl-logistic-services-for-woocommerce/blob/538222ab295ba3d412552e0c4f07b872607de732/pr-dhl-woocommerce/includes/front-end/class-pr-dhl-front-end-paket.php#L385